client/src/tests/ProductDetails/pdrender.test.js

```diff
@@ -1,40 +1,209 @@
+    
+  })
+
+  it('should render sku select', () => {
+    const selectedStyle = ExampleProductStyles.results[0]; 
+    const inStockSkus = Object.entries(selectedStyle?.skus ?? {})?.filter(sku => (sku?.[1]?.quantity ?? 0) > 0);
+    render(<SizeSelector inStockSkus={inStockSkus} selectedSku={inStockSkus[0][0]} showLabel={true} />);
+
+    expect(screen.getByRole('combobox')).toBeTruthy();
+    expect(screen.getAllByRole('option').length).toBe(inStockSkus.length + 1);
+    expect(screen.getByLabelText('Please select size')).toBeTruthy(); 
+
+  });
+
+  it('should render quantity select', () => {
+    const selectedStyle = ExampleProductStyles.results[0]; 
+    const inStockSkus = Object.entries(selectedStyle?.skus ?? {})?.filter(sku => (sku?.[1]?.quantity ?? 0) > 0);
+    const selectedSku = inStockSkus[0]
+    render(<QuantitySelector selectedSku={selectedSku[0]} selectedStyle={selectedStyle} selectedQuantity={undefined} setSelectedQuantity={() => {}}/>)
+    
+    expect(screen.getByRole('combobox')).toBeTruthy();
+    expect(screen.getAllByRole('option').length).toBe(selectedSku[1].quantity + 1);
+    expect(screen.getByLabelText(`select quantity for ${selectedStyle?.name} sku ${selectedSku[0]}`)).toBeTruthy(); 
+
+  })
+
   it('should render cart actions', () => {
-    render(<ProductCartActions />);
+    const selectedStyle = ExampleProductStyles.results[0]; 
+    const inStockSkus = Object.entries(selectedStyle?.skus ?? {})?.filter(sku => (sku?.[1]?.quantity ?? 0) > 0);
+
+    render(<ProductCartActions selectedStyle={selectedStyle} />);
     expect(screen.getAllByRole('button').length).toBe(2);
-    //TODO: complete tests
+    expect(screen.getAllByRole('combobox').length).toBe(2);
+    const addToCartButton = screen.getByText('Add to Cart');
+    expect(addToCartButton).toBeTruthy(); 
+    //click without sku selected
+    fireEvent.click(addToCartButton); 
+    
+    //select a sku
+    const skuSelect = screen.getByText('Select Size');
+    expect(skuSelect).toBeTruthy(); 
+    fireEvent.change(skuSelect, {target: {value: inStockSkus[0][0] }});
+
+    fireEvent.click(addToCartButton); 
+
   });
 
-  it('should render product gallery', () => {
-    render(<ProductGallery />);
-    expect(screen.getByText('Carousel')).toBeTruthy();
-    expect(screen.getByText('Thumbnails')).toBeTruthy();
-    //TODO: complete tests
+  it('should render x product style selects & images', () => {
+    const setSelectedSyle = jest.fn(); 
+    render(<ProductStyleSelector selectedStyle={ExampleProductStyles.results[0]} productStyles={ExampleProductStyles} setSelectedStyle={setSelectedSyle}  />);
+    ExampleProductStyles.results.forEach((style) => {
+      expect(screen.getByLabelText(style.name)).toBeTruthy(); 
+      expect(screen.getByRole('img', {name: `style ${style.name} select preview`})).toBeTruthy();
+    });
+    const radioGroup = screen.getAllByRole('radio'); 
+    expect(radioGroup.length).toBeTruthy(); 
```

Comment:
since this is hard coded data couldnt you say how many radio buttons you expect to have instead of just expecting to have at least one?